homeassistant/components/modbus/binary_sensor.py

```diff
@@ -92,65 +93,29 @@ async def async_setup_platform(
             hub = hass.data[MODBUS_DOMAIN][discovery_info[CONF_NAME]]
         if CONF_SCAN_INTERVAL not in entry:
             entry[CONF_SCAN_INTERVAL] = DEFAULT_SCAN_INTERVAL
-        sensors.append(ModbusBinarySensor(hub, hass, entry))
+        sensors.append(ModbusBinarySensor(hub, entry))
 
     async_add_entities(sensors)
 
 
-class ModbusBinarySensor(BinarySensorEntity):
+class ModbusBinarySensor(basePlatform, BinarySensorEntity):
     """Modbus binary sensor."""
 
-    def __init__(self, hub, hass, entry):
+    def __init__(self, hub: ModbusHub, entry: dict[str, Any]):
         """Initialize the Modbus binary sensor."""
-        self._hub = hub
-        self._hass = hass
-        self._name = entry[CONF_NAME]
-        self._slave = entry.get(CONF_SLAVE)
-        self._address = int(entry[CONF_ADDRESS])
-        self._device_class = entry.get(CONF_DEVICE_CLASS)
-        self._input_type = entry[CONF_INPUT_TYPE]
-        self._value = None
-        self._available = True
-        self._scan_interval = timedelta(seconds=entry[CONF_SCAN_INTERVAL])
+        basePlatform.__init__(self, hub, entry)
```

Comment:
```suggestion
        super().__init__(hub, entry)
```

Reply:
I did it to be sure the right __init__ was called (I have had some bad experience with super() and multiple inheritance).

Reply:
Done.